### ISSUE TITLE
Better handle mock output factory

### ISSUE THREAD

**Issue by BlankDev117** [OpenSourceKingdom/OSK.Functions.Outputs#7]
Current implementation prohibits easy access to an implementation of output factory that allows for accessibility for mock/unit tests. This should be changed so that the project can better support unit testing
